packages/app-desktop/gui/NoteEditor/NoteBody/TinyMCE/utils/openEditDialog.ts

```diff
@@ -95,7 +95,7 @@ export default function openEditDialog(editor: any, markupToHtml: any, dispatchD
 		// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
 		onSubmit: async (dialogApi: any) => {
 			const newSource = newBlockSource(dialogApi.getData().languageInput, dialogApi.getData().codeTextArea, source);
-			const md = `${newSource.openCharacters}${newSource.content.trim()}${newSource.closeCharacters}`;
+			const md = `${newSource.openCharacters}${newSource.content}${newSource.closeCharacters}`;
```

Comment:
> I am not entirely sure why it is in there in the first place nor can I think of a reason why markdown would need to be trimmed if the user wrote markdown with trailing whitespace in our use case, so I removed it.

It looks like this was added in https://github.com/laurent22/joplin/commit/84c3ef144d95bbe86f7a1655b56b927e0d1d7636#diff-4eba09d3f5fecc9c57315952b11b3ab68a356d30e87314712467c7e8972ccaa1R232 (the same commit that added the rich text editor).

In the past, we've had bugs related to extra lines being added to the end of code and math regions when saving content in the Rich Text editor (e.g. https://github.com/laurent22/joplin/issues/6822, https://github.com/laurent22/joplin/issues/9589). Perhaps this was an attempted/partial fix for one of those issues?

Reply:
Sounds great. I've also had a quick look at the issues you linked and tested those issues and verified that those issues are separate from the one this PR is aiming to fix, as the code block's onSubmit callback is not called when changing editors.

Reply:
Yes there was a problem before where newlines kept being added each time the dialog was closed, but maybe it's no longer happening. Let's merge for now and see if anyone finds any issue.